stan/math/prim/prob/poisson_log_lpmf.hpp

```diff
@@ -20,26 +20,23 @@ template <bool propto, typename T_n, typename T_log_rate>
 return_type_t<T_log_rate> poisson_log_lpmf(const T_n& n,
                                            const T_log_rate& alpha) {
   using T_partials_return = partials_return_t<T_n, T_log_rate>;
-
-  static const char* function = "poisson_log_lpmf";
-
   using std::exp;
-
-  if (size_zero(n, alpha)) {
-    return 0.0;
-  }
-
-  T_partials_return logp(0.0);
-
+  static const char* function = "poisson_log_lpmf";
   check_nonnegative(function, "Random variable", n);
   check_not_nan(function, "Log rate parameter", alpha);
   check_consistent_sizes(function, "Random variable", n, "Log rate parameter",
                          alpha);
 
+  if (size_zero(n, alpha)) {
+    return 0.0;
+  }
   if (!include_summand<propto, T_log_rate>::value) {
     return 0.0;
   }
 
+  T_partials_return logp(0.0);
```

Comment:
[optional]
Same deal---move this to before where it gets used.  The point is that you don't want to have to scan for where a variable is initialized when using it.